lib/middleware.js

```diff
@@ -132,33 +132,34 @@ exports.isFeatureEnabled = function (featureName) {
 
 exports.checkQuota = function () {
   return function (req, res, next) {
-    if (utils.isServiceFabrikOperation(req.body) || (CONST.HTTP_METHOD.PATCH === req.method && utils.isNotPlanUpdate(req.body))) {
-      logger.debug('[Quota]: Check skipped as it is ServiceFabrikOperation or a normal instance update ...: calling next handler..');
+    if (utils.isServiceFabrikOperation(req.body) || (CONST.HTTP_METHOD.PATCH === req.method && (utils.isNotPlanUpdate(req.body) || utils.isSameSkuUpdate(req.body, catalog)))) {
+      logger.debug('[Quota]: Check skipped as it is ServiceFabrikOperation or a normal instance update or plan update with same sku...: calling next handler..');
       next();
     } else {
       const platform = _.get(req, 'body.context.platform');
       if (platform === CONST.PLATFORM.CF) {
         const orgId = req.body.organization_guid || req.body.context.organization_guid || req.body.previous_values.organization_id;
         if (orgId === undefined) {    
           next(new BadRequest(`organization_id is undefined`));   
+        } else {
```

Comment:
I see that QuotaManager also has the logic to skip quota based on whitelisted org or not. So shouldn't we move the above checks that are in `if` clause apart from serviceFabrikOperation operation into the `quotaManager` so that we avoid all these util functions. To me util functions are something reusable but in this case i am seeing what should ideally be part of quota manager logic is getting spilled into utils. would recommend keeping everything within QuotaManager. (Maybe SF operation check is an exception and can be kept outside as is but others i think must be encapsulated within QM class) 

Also figuring out skuname is part of quota manager. So now thats duped & also spilled onto utils.